### PR TITLE
RavenDB-18675 Corax's IndexWriter cleanup.

### PR DESCRIPTION
**Pull request by maciejaszyk**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-18675 

### Additional description

We had multiple classes/structs inside `IndexWriter` that could be moved outside to increase the readability of the writer itself.

### How risky is the change?

- Not relevant

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

**Review by arekpalinski — APPROVED** [review]

File: src/Corax/IndexWriter/EntriesModifications.cs [hunk 1]
```diff
@@ -0,0 +1,354 @@
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+using System.Text;
+using Corax.Utils;
+using Sparrow.Server;
+using Voron.Util;
+
+namespace Corax.IndexWriter;
```
Comment (ayende) [inline src/Corax/IndexWriter/EntriesModifications.cs]: The namespace and the class name being the same is going to be pretty confusing.
Let's call it `Corax.IndexWriting` and then have `Corax.IndexWriting.IndexWriter` ?

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Corax/IndexWriter/EntriesModifications.cs (hunk 1 above)
Comment (redknightlois) [reply src/Corax/IndexWriter/EntriesModifications.cs]: Maybe too late, but if we are going to be doing cleanup maybe we could do:  `Corax.Indexing` and `Corax.Searching` or `Corax.Querying` and unify the `IndexSearcher` and the `Corax.Queries` under that.

Comment (arekpalinski) [reply src/Corax/IndexWriter/EntriesModifications.cs]: I like the idea of `Corax.Indexing` and `Corax.Querying` namespaces. @maciejaszyk what do you think?

Comment (maciejaszyk) [reply src/Corax/IndexWriter/EntriesModifications.cs]: Done :)

**Pull request merged**